toolchain/check/pattern_match.cpp

```diff
@@ -164,17 +122,25 @@ class MatchContext {
 // might emit insts.
 // TODO: Require that `entry.scrutinee_id` is valid if and only if insts should
 // be emitted, once we start emitting `Param` insts in the `ParamPattern` case.
-auto EmitPatternMatch(Context& context, MatchContext& match,
-                      MatchContext::WorkItem entry) -> void {
+auto MatchContext::EmitPatternMatch(Context& context,
```

Comment:
Should the function comment be moved to the declaration, instead of leaving it on the definition?

Reply:
Since it's a private method, I think it can validly go in either place. I feel like readers are more likely to be looking at the function body than the callsite, so putting it here is probably more helpful, but I don't feel too strongly.

Reply:
For me, the function is non-obvious in what it does. Maybe I'm misunderstanding [style](https://google.github.io/styleguide/cppguide.html#Function_Comments), but it says "Declaration comments describe use of the function (when it is non-obvious); comments at the definition of a function describe operation." Is there an exception for private methods that I'm missing?

Reply:
Huh, I could've sworn the style guide allowed the API comment to go on the definition for private methods, but you're right. Fixed.